src/parse/lexer.rs

```diff
@@ -550,6 +554,7 @@ where
             if let Some(c) = self.take_number(radix) {
                 value_text.push(c);
             } else if self.chr0 == Some('_') && Lexer::<T>::is_digit_of_radix(self.chr1, radix) {
+                value_text.push('_');
```

Comment:
lol oops.

Reply:
I think that should be a lex error, having it become an underscore would probably result in a more confusing error.